src/config/client_config.cc

```diff
@@ -0,0 +1,159 @@
+/*GRB*
+
+    Gerbera - https://gerbera.io/
+    
+    client_config.h - this file is part of Gerbera.
+    
+    Copyright (C) 2020 Gerbera Contributors
+    
+    Gerbera is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License version 2
+    as published by the Free Software Foundation.
+
+    Gerbera is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Gerbera.  If not, see <http://www.gnu.org/licenses/>.
+
+    $Id$
+*/
+
+/// \file client_config.cc
+
+#include "client_config.h" // API
+#include "content_manager.h"
+#include "util/upnp_clients.h"
+
+#include <utility>
+
+ClientConfig::ClientConfig()
+{
+    clientInfo.flags = 0;
+    ip = nullptr;
+    userAgent = nullptr;
+    clientInfo.type = ClientType::Unknown;
+}
+
+ClientConfig::ClientConfig(int flags, std::string ip, std::string userAgent, ClientType clientType)
+    : ip(ip)
+    , userAgent(userAgent)
+{
+    clientInfo.type = clientType;
+    clientInfo.flags = flags;
+    clientInfo.name = fmt::format("Manual Setup for{}{}", !ip.empty() ? " IP " + ip : "", !userAgent.empty() ? " UserAgent " + userAgent : "");
+}
+
+ClientConfigList::ClientConfigList()
+{
+}
+
+void ClientConfigList::add(const std::shared_ptr<ClientConfig>& client)
+{
+    AutoLock lock(mutex);
+    _add(client);
+}
+
+void ClientConfigList::_add(const std::shared_ptr<ClientConfig>& client)
+{
+    list.push_back(client);
+}
+
+std::vector<std::shared_ptr<ClientConfig>> ClientConfigList::getArrayCopy()
+{
+    AutoLock lock(mutex);
+    return list;
+}
+
+std::shared_ptr<ClientConfig> ClientConfigList::get(size_t id)
+{
+    AutoLock lock(mutex);
+
+    if (id >= list.size())
+        return nullptr;
+
+    return list[id];
+}
+
+void ClientConfigList::remove(size_t id)
+{
+    AutoLock lock(mutex);
+
+    if (id >= list.size()) {
+        log_debug("No such ID {}!", id);
+        return;
+    }
+
+    list.erase(list.begin() + id);
+    log_debug("ID {} removed!", id);
+}
+
+std::string ClientConfig::mapClientType(ClientType clientType)
```

Comment:
dont use ClientType, please use quirk flags

Reply:
Kept the conversion function for completeness